swaylock/main.c

```diff
@@ -115,14 +115,100 @@ void notify_key(enum wl_keyboard_key_state state, xkb_keysym_t sym, uint32_t cod
 	}
 }
 
+static void cairo_set_source_u32(cairo_t *cairo, uint32_t color) {
```

Comment:
If you move it to `client/cairo.h` which includes `gdk-pixbuf/gdk-pixbuf.h`, then code that uses that function also needs to be able to include gdk-pixbuf. It doesn't need to link against it, but when building, it needs to find that header.